albogdan/skule-vote
# Fix for Invalid Date throwing error 500

## Run `python manage.py migrate` before reviewing this PR.

## Overview

- Resolves #111
- Adds a check to ensure a valid date and valid time is entered when creating a new `ElectionSession`
- Updated the help text of the `start_time` and `end_time` of an `ElectionSession` to explain using a format of `YYYY-MM-DD and HH:MM:SS`

## Unit Tests Created

- Adds a corresponding unit test that checks cases for this for start and end dates and start and end times.


## Steps to QA

- Try creating a new `ElectionSession` with invalid start date/end date/start time/end time. 
- You should get invalid messages for `Enter a valid date.` for the dates, `Enter a valid time.` for the times. As well, at the top you should get an overall message of `The ElectionSession must have a valid start and end time.` 
- If the dates and times actually exist then the `ElectionSession` should create successfully.
- Play around, see if I overlooked anything.
- Make sure that the help text for `start_time` and `end_time` of an `ElectionSession` shows `YYYY-MM-DD and HH:MM:SS`
